### PR TITLE
Add feature to clone sites

### PR DESCRIPTION
**Pull request by mfairchild365**

### CODE REVIEW

File: sites/all/modules/unl_multisite/cron.php [hunk 1]
```diff
@@ -271,15 +271,16 @@ function unl_add_site($site_path, $uri, $clean_url, $db_prefix, $site_id) {
...
-          ;
+    . '://' . $database['username']
+    . ':'   . $database['password']
+    . '@'   . $database['host']
+    . ($database['port'] ? ':' . $database['port'] : '')
+    . '/'   . $database['database']
+  ;
   $db_prefix .= '_' . $database['prefix'];
 
   $php_path = escapeshellarg($_SERVER['_']);
```
Comment (ericras) [inline sites/all/modules/unl_multisite/cron.php]: FWIW, I've found I needed to use PHP_BINARY to debug in phpstorm

File: sites/all/modules/unl_multisite/cron.php [hunk 2]
```diff
@@ -297,19 +298,109 @@ function unl_add_site($site_path, $uri, $clean_url, $db_prefix, $site_id) {
...
+  }
+}
+
+function getTableNamesWithPrefix($prefix) {
+  $database = $GLOBALS['databases']['default']['default'];
+
+  $query = "";
+  $query .= "SELECT `table_name`";
+  $query .= " FROM INFORMATION_SCHEMA.TABLES";
+  $query .= " WHERE `table_schema` = '".db_escape_table($database['database'])."'";
```
Comment (ericras) [inline sites/all/modules/unl_multisite/cron.php]: My database name is UNL-CMS and this blows away the -

$query .= " WHERE `table_schema` = `".$database['database']."`";

File: sites/all/modules/unl_multisite/cron.php [hunk 3]
```diff
@@ -297,19 +298,109 @@ function unl_add_site($site_path, $uri, $clean_url, $db_prefix, $site_id) {
...
+  $from_db_prefix = $clone_from_site['db_prefix'] .'_' . $database['prefix'];
+  $to_db_prefix = $clone_to_site['db_prefix'] .'_' . $database['prefix'];
+  
+  // Grab the list of tables we need to clone.
+  $tables = getTableNamesWithPrefix($from_db_prefix);
+
+  // Clone the site's tables
+  foreach ($tables as $table) {
+    $from_table = $table;
+    $to_table = str_replace_once($from_db_prefix, $to_db_prefix, $table);
```
Comment (ericras) [inline sites/all/modules/unl_multisite/cron.php]: str_replace_once ?

Comment (mfairchild365) [inline sites/all/modules/unl_multisite/cron.php]: it is a function, just not one that you have. :P This should be fixed now, and perhaps I shouldn't blindly auto-complete function names in the future.

File: sites/all/modules/unl_multisite/cron.php (hunk 2 above)
Comment (mfairchild365) [inline sites/all/modules/unl_multisite/cron.php]: Just to be super cautious, I left it in there but am now escaping it The Right Way™

**Pull request merged**